databricks/koalas/generic.py

```diff
@@ -1279,6 +1279,8 @@ def groupby(self, by, as_index: bool = True):
             col_by = [_resolve_col(df, col_or_s) for col_or_s in by]
             return DataFrameGroupBy(df_or_s, col_by, as_index=as_index)
         if isinstance(df_or_s, Series):
+            if not isinstance(by[0], Series):
```

Comment:
I think you should remove `if isinstance(by, str):`,.
We also should fix the error message `raise ValueError('Not a valid index: TODO')` to match with pandas'

Reply:
i think maybe we hard to remove `if isinstance(by, str):` , since `str` is valid for `DataFrameGroupBy` ??

```python
>>> pdf
      regiment   company experience      name  preTestScore  postTestScore
0   Nighthawks  infantry    veteran    Miller             4             25
1   Nighthawks  infantry     rookie  Jacobson            24             94
2   Nighthawks   cavalry    veteran       Ali            31             57
3   Nighthawks   cavalry     rookie    Milner             2             62
4     Dragoons  infantry    veteran     Cooze             3             70
5     Dragoons  infantry     rookie     Jacon             4             25
6     Dragoons   cavalry    veteran    Ryaner            24             94
7     Dragoons   cavalry     rookie      Sone            31             57
8       Scouts  infantry    veteran     Sloan             2             62
9       Scouts  infantry     rookie     Piger             3             70
10      Scouts   cavalry    veteran     Riani             2             62
11      Scouts   cavalry     rookie       Ali             3             70
>>> pdf.groupby('company')
<pandas.core.groupby.generic.DataFrameGroupBy object at 0x124f39780>
```